src/test/sql/sql_translator_test.cpp

```diff
@@ -536,4 +538,44 @@ TEST_F(SQLTranslatorTest, MixedAggregateAndGroupBySelectList) {
   ASSERT_STORED_TABLE_NODE(aggregate_node->left_child()->right_child(), "table_a");
 }
 
+TEST_F(SQLTranslatorTest, CreateView) {
+  const auto query = "CREATE VIEW my_first_view AS SELECT * FROM table_a WHERE a = 'b';";
+  auto result_node = compile_query(query);
+  EXPECT_EQ(result_node->type(), LQPNodeType::CreateView);
+  auto create_view_node = std::dynamic_pointer_cast<CreateViewNode>(result_node);
+
+  EXPECT_EQ(create_view_node->type(), LQPNodeType::CreateView);
+  EXPECT_EQ(create_view_node->view_name(), "my_first_view");
+
+  auto view_lqp = create_view_node->lqp();
+  EXPECT_EQ(view_lqp->type(), LQPNodeType::Projection);
+  EXPECT_FALSE(view_lqp->right_child());
+
+  auto ts_node_1 = std::static_pointer_cast<PredicateNode>(view_lqp->left_child());
```

Comment:
Just to point this out: Here we have 6 clustered lines of code checking whether the node is a PredicateNode with specific params. One EXPECT_PREDICATE_NODE line would be much more intuitive, imo

Reply:
Then you would need one `EXPECT_..._NODE` for each node type. This is not really generic. I would rather create a second PredicateNode with the actual constructor interface and compare those two.